### PR TITLE
Upgrade Hunter to get newer nlohmann_json

### PR DESCRIPTION
**Pull request by gumb0**
Required for #716 

Older version of nlohmann_json has some compilation errors when `<any>` is included, see https://github.com/wasmx/fizzy/pull/716#issuecomment-779365591 for links.

### CODE REVIEW

File: cmake/Hunter/config.cmake [hunk 1]
```diff
@@ -2,6 +2,4 @@
 hunter_config(
     benchmark
     VERSION 1.5.0
-    URL https://github.com/google/benchmark/archive/v1.5.0.tar.gz
```
Comment (chfast) [inline cmake/Hunter/config.cmake:5]: Why is this removed? If we get benchmark 1.5.0 by default you can remove all of it.

Comment (gumb0) [reply cmake/Hunter/config.cmake:5]: Yes, it's default.

Should I leave this file empty? It complains if I delete it:
```
-- Build type: Debug

[hunter ** FATAL ERROR **] Config not found: /home/andrei/dev/fizzy/cmake/Hunter/config.cmake
[hunter ** FATAL ERROR **] [Directory:/home/andrei/dev/fizzy/test]

------------------------------ ERROR -----------------------------
    https://hunter.readthedocs.io/en/latest/reference/errors/error.incorrect.input.data.html
------------------------------------------------------------------

CMake Error at /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_error_page.cmake:12 (message):
Call Stack (most recent call first):
  /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_fatal_error.cmake:20 (hunter_error_page)
  /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_user_error.cmake:7 (hunter_fatal_error)
  /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_set_config_location.cmake:50 (hunter_user_error)
  /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_apply_gate_settings.cmake:85 (hunter_set_config_location)
  /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_finalize.cmake:73 (hunter_apply_gate_settings)
  /home/andrei/.hunter/_Base/Download/Hunter/0.23.294/0dd1ee8/Unpacked/cmake/modules/hunter_add_package.cmake:23 (hunter_finalize)
  test/CMakeLists.txt:16 (hunter_add_package)


-- Configuring incomplete, errors occurred!
See also "/home/andrei/dev/fizzy/build/CMakeFiles/CMakeOutput.log".
See also "/home/andrei/dev/fizzy/build/CMakeFiles/CMakeError.log".
```

Comment (chfast) [reply cmake/Hunter/config.cmake:5]: You can remove the file if you also remove the `LOCAL` option from `HunterGate()`.

**Review by axic — APPROVED** [review]

File: cmake/Hunter/config.cmake (hunk 1 above)
Comment (gumb0) [reply cmake/Hunter/config.cmake:5]: Done, thanks.

**Review by chfast — APPROVED** [review]
